src/utils.ts

```diff
@@ -127,41 +127,53 @@ export const websocketStream = (
 /**
  * Encodes URI components and returns a URL.
  * @param baseUrl base URL
- * @param params (optional) parameters for queries
+ * @param params (optional) path params for the url
+ * @param queryParams (optional) query params for the url
  */
-export const buildUrl = (baseUrl: string, params?: any[] | object): string => {
+export const buildUrl = (
+  baseUrl: string,
+  params?: any[],
```

Comment:
we should not force people to use `params` to use `queryParams`. One way would be to check, what we got for each variable and just treat arras as `/sdf/sfd/` and an object as `?sdf=sf`